### PR TITLE
build: Add Zig minimal example

### PR DESCRIPTION
**Pull request by vanillajonathan**
This is a Zig example which uses the `libprql_lib.h` C header file and loads the PRQL library using dynamic linking.

### CODE REVIEW

**Comment by max-sixty** [comment]
Cool!

Could we add a small test? Otherwise it's likely to decay.

(an alternative would be posting it to a gist / a blog, if you prefer to just publish some code quickly)

**Comment by vanillajonathan** [comment]
I have now added a unit test.

**Comment by max-sixty** [comment]
Yes I also mean to _run_ the test in CI...

I think that could be as small as just adding a job to https://github.com/PRQL/prql/blob/f9a983d1bc0d139b94246aa240d2c59edc35d709/.github/workflows/test-lib.yaml, which is easier that making a new workflow...

File: .github/workflows/test-lib.yaml [hunk 1]
```diff
@@ -34,3 +34,11 @@ jobs:
       - name: Run basic C++ example
         working-directory: bindings/prql-lib/examples/minimal-cpp
         run: make run
+      - name: 🔧 Setup Zig
```
Comment (max-sixty) [inline .github/workflows/test-lib.yaml:55]: Let's do a separate job within this workflow though, so it's clearly delineated, particularly when there's a failure. You can copy & paste most things up to this line.

Comment (vanillajonathan) [reply .github/workflows/test-lib.yaml:55]: Okay, I've done that now.

**Comment by max-sixty** [comment]
Looks great, thanks @vanillajonathan !

**Comment by max-sixty** [comment]
Thanks @vanillajonathan !

**Pull request merged**